dbms/src/Operators/HashJoinProbeTransformOp.cpp

```diff
@@ -251,7 +220,7 @@ OperatorStatus HashJoinProbeTransformOp::executeIOImpl()
         probe_transform->flushMarkedSpillData();
         probe_transform->finalizeProbe();
         switchStatus(ProbeStatus::WAIT_PROBE_FINISH);
-        return OperatorStatus::WAITING;
+        return OperatorStatus::HAS_OUTPUT;
```

Comment:
Why original code return `WAITING`?

Reply:
Because later, we need to poll in the wait reactor for other task probes to finish before continuing with the restore build.
Now, return to the CPU thread pool to trigger wait-for-notify.

Reply:
you mean return to cpu thread pool, if `isProbeFinishedForPipeline` return false, it will go to `wait-for-notify` state?

Reply:
yes

Reply:
`onWaitProbeFinishDone` will be called instead of triggering wait-for-notify because `finalizeProbe` is called.

Reply:
> `onWaitProbeFinishDone` will be called instead of triggering wait-for-notify because `finalizeProbe` is called.

yes, you are right.